src/components/Courses/Courses.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import './Courses.css';
+import PropTypes from 'prop-types';
+import SearchBar from './components/SearchBar/SearchBar';
+import CourseCard from './components/CourseCard/CourseCard';
+import Button from '../../common/Button/Button';
+
+const Courses = ({ courses, authors, onAddCourseClick }) => {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const handleSearch = (term) => {
+    setSearchTerm(term);
+  };
+
+  const filteredCourses = courses.filter(course =>
```

Comment:
Optional, so feel free to skip, but here's a good place where you could utilize `useMemo` hook to improve performace of this component.